src/Compilers/VisualBasic/Test/Semantic/Semantics/RequiredMembersTests.vb

```diff
@@ -0,0 +1,1283 @@
+    extends [mscorlib]System.ValueType
+{
+    .custom instance void [mscorlib]System.Runtime.CompilerServices.RequiredMemberAttribute::.ctor() = (
+        01 00 00 00
+    )
+    .field private int32 f
+
+    .method family hidebysig specialname rtspecialname 
+        instance void .ctor () cil managed 
+    {
+        .custom instance void [mscorlib]System.ObsoleteAttribute::.ctor(string, bool) = (
+            01 00 5f 43 6f 6e 73 74 72 75 63 74 6f 72 73 20
+            6f 66 20 74 79 70 65 73 20 77 69 74 68 20 72 65
+            71 75 69 72 65 64 20 6d 65 6d 62 65 72 73 20 61
+            72 65 20 6e 6f 74 20 73 75 70 70 6f 72 74 65 64
+            20 69 6e 20 74 68 69 73 20 76 65 72 73 69 6f 6e
+            20 6f 66 20 79 6f 75 72 20 63 6f 6d 70 69 6c 65
+            72 2e 01 00 00
+        )
+        .custom instance void [mscorlib]System.Runtime.CompilerServices.CompilerFeatureRequiredAttribute::.ctor(string) = (
+            01 00 0f 52 65 71 75 69 72 65 64 4d 65 6d 62 65
+            72 73 00 00
+        )
+        ret
+    }
+
+    .method public hidebysig specialname 
+        instance int32 get_Prop () cil managed 
+    {
+        ldarg.0
+        ldfld int32 S::f
+        ret
+    }
+
+    .method public hidebysig specialname 
+        instance void set_Prop (
+            int32 'value'
+        ) cil managed 
+    {
+        ldarg.0
+        ldarg.1
+        stfld int32 S::f
+        ret
+    }
+
+    .property instance int32 Prop()
+    {
+        .custom instance void [mscorlib]System.Runtime.CompilerServices.RequiredMemberAttribute::.ctor() = (
+            01 00 00 00
+        )
+        .get instance int32 S::get_Prop()
+        .set instance void S::set_Prop(int32)
+    }
+}
+"
+
+            Dim ilAssembly = CompileIL(il)
+
+            Dim comp = CreateCompilation("
+Module M
+    Sub Main()
+        Dim s = New S()
+    End Sub
+End Module
+", {ilAssembly}, targetFramework:=TargetFramework.Net70)
+
+            comp.AssertTheseDiagnostics(<expected>
+BC37321: Required member 'Public Overloads Property Prop As Integer' must be set in the object initializer or attribute arguments.
+        Dim s = New S()
+                    ~
+                                        </expected>)
+        End Sub
+
+        <Fact>
+        Public Sub RequiredMemberAttributeDisallowedInSource()
+            Dim comp = CreateCompilation("
+Imports System.Runtime.CompilerServices
+<RequiredMember>
+Public Class C
+    <RequiredMember>
+    Public Property P As Integer
+
+    <RequiredMember>
+    Public F As Integer
+End Class", targetFramework:=TargetFramework.Net70)
+
+            comp.AssertTheseDiagnostics(<expected><![CDATA[
+BC37325: 'System.Runtime.CompilerServices.RequiredMemberAttribute' is reserved for compiler usage only.
+<RequiredMember>
+ ~~~~~~~~~~~~~~
+BC37325: 'System.Runtime.CompilerServices.RequiredMemberAttribute' is reserved for compiler usage only.
+    <RequiredMember>
```

Comment:
><RequiredMember> [](http://example.com/codeflow?start=4&length=16)

Consider adding a comment in source to distinguish this location from the one below.